src/Tools/dotnet-monitor/CollectionRules/ExecuteAction.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules
+{
+    internal sealed class ExecuteAction : IAction<ExecuteOptions>
+    {
+        public async Task<ActionResponse> Execute(ExecuteOptions options, CancellationToken token)
+        {
+            ActionResponse executeResponse = new();
+
+            await Task.Run(() =>
+            {
+                string path = options.Path;
+                string arguments = options.Arguments;
+
+                Process process = Process.Start(path, arguments);
+
+                process.WaitForExit();
```

Comment:
Do we have a timeout we can plugin to the action?

Reply:
I wonder if all actions should have this and the execution pipeline can handle the timeout by raising cancellation on the cancellation token. We'd probably want this to be a configurable property on every action. We might need to think about the design on this one for a little bit to strike a balance on where this information is found in configuration and who is responsible for binding the information into some set of options (is there an overlay options that we bind for each action section? Do we force action options to implement an interface? etc).